Tests/Invoke-Tests.ps1

```diff
@@ -0,0 +1,54 @@
+<#
+.SYNOPSIS
+Runner to invoke Acceptance, Quality and / or Unit tests
+
+.DESCRIPTION
+Test wrapper that invokes
+
+.PARAMETER TestType
+[Optional] The type of test that will be executed. The parameter value can be either All (default), Acceptance, Quality or Unit
+
+.EXAMPLE
+Invoke-AcceptanceTests.ps1
+
+.EXAMPLE
+Invoke-AcceptanceTests.ps1 -TestType Unit
+
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("All", "Acceptance", "Quality", "Unit")]
+    [String] $TestType = "All"
+)
+
+$TestParameters = @{
+    OutputFormat = 'NUnitXml'
+    OutputFile   = "$PSScriptRoot\TEST-$TestType.xml"
+    Script       = "$PSScriptRoot"
+    PassThru     = $True
+}
+if ($TestType -ne 'All') {
+    $TestParameters['Tag'] = $TestType
+}
+
+# Install required modules
+$RequiredModules = @("Pester", "PSScriptAnalyzer")
+foreach ($Module in $RequiredModules) {
+    if (!(Get-Module -Name $Module -ListAvailable)) {
+        Write-Output "Installing $Module"
+        Install-Module -Name $Module -Scope CurrentUser -Force
```

Comment:
Personally I'm not sure whether scripts should really be managing module installations. I would opt for using [Requires attributes](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_requires?view=powershell-6) and having the script runner be responsible for installing required modules. Up for discussion either way.

Reply:
Will take a look into this 👍

Reply:
I've updated Invoke-Tests.ps1 to use #Requires and works perfectly in the build. Looks much neater and readable, rather than a foreach loop. I'd like to use #Requires moving forward.